### ISSUE TITLE
2016.11.1 :[ERROR][11182] Failed to import beacons avahi_announce

### ISSUE THREAD

**Issue by rukender** [saltstack/salt#38684]
### Description of Issue/Question
I'm getting this error in most of the minion which are upgraded with the latest version i.e. 2016.11.1
```
2017-01-11 06:39:47,303 [salt.loader      ][ERROR   ][11182] Failed to import beacons avahi_announce, this is due most likely to a syntax error:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1332, in _load_module
    ), fn_, fpath, desc)
  File "/usr/lib/python2.7/site-packages/salt/beacons/avahi_announce.py", line 30, in <module>
    SERVER = dbus.Interface(BUS.get_object(avahi.DBUS_NAME, avahi.DBUS_PATH_SERVER),
NameError: name 'avahi' is not defined
```
### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 6.8 Final
        machine: x86_64
        release: 2.6.32-642.11.1.el6.x86_64
         system: Linux
        version: CentOS 6.8 Final
```

**Comment by rukender** [comment]
I can confirm that this error is because of the upgrade only, here is the link which gives information about which all files updated in 2016.11.1 version.
https://www.archlinux.org/packages/community/any/salt/files/

@SaltStackSupport please provide some solution asap.

Thanks

**Comment by Ch3LL** [comment]
@rukender just to confirm are you trying to use the avahi beacon? or is this showing without attempting to configure the beacon?

If you are trying to use avahi beacon cna you confirm you have the following installed:

```
python-avahi
dbus-python
```

My guess is you do not have avahi installed hence the nameerror. You can confirm whether you have avahi properlly installed by running `python -c "import avahi"` and see if it errors out.

**Comment by rukender** [comment]
@Ch3LL ,

I can see this package is not installed. 
Also, pyhton avahi module came after upgrading to the latest version of salt-minion i.e. 2016.11.1

I'm getting error while running python -c "import avahi"

output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named avahi

So to resolve this issue, either we need to install this avahi package or remove it right? if we need to remove this module could you please let me know how to do it. 

Thanks
Rukender

**Comment by Ch3LL** [comment]
@rukender are you trying to use the avahi beacon though? If you are trying to use the avahi beacon then yes please install the package if you have no intention of using this beacon then this is an issue for this module to be erroring out like that and we need to fix this so it does not error when attempting to load the beacon.

What does your minion config look like so I can possibly replicate this issue?

**Comment by rukender** [comment]
@Ch3LL , we are not planning to use avahi beacon. 

So, could you please help me fix this issue. What exactly you want to know from minion config file. 

Thanks

**Comment by rallytime** [comment]
@rukender If you change [this line](https://github.com/saltstack/salt/blob/cfd82d16312a325c895f6d3ed23dc40da1704a3e/salt/beacons/avahi_announce.py#L35) to include the `NameError`, this will resolve the issue for you:

Change this:
```
except ImportError:
```
to this:
```
except (ImportError, NameError):
```
I'll submit a fix for this right now.

**Comment by rukender** [comment]
@rallytime, thank you for the solution.

May I know if I change except ImportError: to except (ImportError, NameError): what it will do ?

Thanks

**Comment by rukender** [comment]
Also, I have replaced 35 line under avahi_announce.py file. 
This is the right position, right?

**Comment by rallytime** [comment]
@rukender Yes, that is the right line.

Basically what is happening here is that since you have the dbus package installed, but not the avahi package, when the loader enters that try/catch block, the `import dbus` call succeeds. But since you don't have the avahi package installed, the stacktrace you posted above occurs when it hits the `avahi.DBUS_NAME` call a couple of lines down.

Adding the `NameError` to the except line catches the NameError exception and sets the `HAS_DBUS` value to False. You should no longer see the stacktrace in your logs. 

I have fixed this with pull request #38713, which will be available in the 2016.11.2 release.
